File: frontend-server/src/app/models/facility.ts
export class Facility {
    private _id: number
    private _name: string
    private _description: string

    public constructor(id: number, name: string, description: string) {
        this._id = id
        this._name = name
        this._description = description
    }

    get id(): number {
        return this._id
    }

    set id(id: number) {
        this._id = id
    }

    get name(): number {
        return this.name
    }

    set name(name: number) {
        this.name = name
    }

    get description(): string {
        return this._description
    }

    set description(description: string) {
        this._description = description
    }
}

File: backend-server/src/main/java/com/unn/model/Facility.java
package com.unn.model;

import java.util.List;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
@Table(name = "s_facility")
@DynamicUpdate
public class Facility {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonInclude(Include.NON_NULL)
    private Long id;

    private String name;
    private String description;

    @OneToMany(mappedBy = "facility")
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JsonIgnore
    private List<Doctor> doctors;
}
